### PR TITLE
Adds secrets field to ListSnapshotsRequest

### PR DESCRIPTION
**Pull request by hakanmemisoglu**
There is no way to directly pass credentials to ListSnapshots call.
This change will add a field required to pass credentials by the
caller.

### CODE REVIEW

**Comment by hakanmemisoglu** [comment]
#370

**Comment by hakanmemisoglu** [comment]
@saad-ali Do we need to update golang version in Travis? (or enable go module)

File: spec.md [hunk 1]
```diff
@@ -1737,6 +1737,12 @@ message ListSnapshotsRequest {
   // and will not block if the snapshot is being processed after
   // it is cut.
   string snapshot_id = 4;
+
+  // Secrets necessary for listing snapshots.
+  // This field is OPTIONAL. It can be used to pass secrets
+  // required to access some subset of the snapshots in the
```
Comment (xing-yang) [inline spec.md]: What do you mean by "some subset of the snapshots"?  How is a subset defined?

Comment (hakanmemisoglu) [reply spec.md]: How about `... access the snapshots...`?

**Comment by jieyu** [comment]
Yeah, we should bump the golang in CI. It's still using 1.10.4

**Comment by hakanmemisoglu** [comment]
@jieyu Sure. I will push a separate PR.

File: spec.md (hunk 1 above)
Comment (saad-ali) [reply spec.md]: How about:

```
  // Secrets required by plugin to complete ListSnapshot request.
  // This field is OPTIONAL. Refer to the `Secrets Requirements`
  // section on how to use this field.
```

**Comment by hakanmemisoglu** [comment]
@saad-ali CI passed.

**Pull request merged**